spot_driver/src/images/spot_image_publisher.cpp

```diff
@@ -90,9 +90,18 @@ bool SpotImagePublisher::initialize() {
   const auto uncompress_images = parameters_->getUncompressImages();
   const auto publish_compressed_images = parameters_->getPublishCompressedImages();
 
+  std::set<spot_ros2::SpotCamera> cameras_used;
+  const auto cameras_used_parameter = parameters_->getCamerasUsed(has_arm_);
+  if (cameras_used_parameter.has_value()) {
+    cameras_used = cameras_used_parameter.value();
+  } else {
+    logger_->logWarn("Invalid cameras_used parameter! Defaulting to publishing from all cameras.");
+    cameras_used = parameters_->getDefaultCamerasUsed(has_arm_);
+  }
+
   // Generate the set of image sources based on which cameras the user has requested that we publish
   const auto sources =
-      createImageSources(publish_rgb_images, publish_depth_images, publish_depth_registered_images, has_arm_);
+      createImageSources(publish_rgb_images, publish_depth_images, publish_depth_registered_images, cameras_used);
 
   // Generate the image request message to capture the data from the specified image sources
   image_request_message_ = createImageRequest(sources, has_rgb_cameras, rgb_image_quality, publish_raw_rgb_cameras);
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 134]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Another thing to be aware of -- the static transforms associated with unpublished camera sources will not be updated. This can potentially lead to disconnected tf trees. There is a ticket for this somewhere (that I cannot find) but I think this should be addressed in a later PR <!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/423?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Why do we care about transforms for unpublished cameras?

Reply:
One place i saw this in testing was that the object detection node looks for fiducials in all of the cameras. So if i only enabled the front cameras but there was a fiducial visible from the back camera you couldn't get the transform to it. But I totally agree with you, I am not sure why you would care about this transform if you didn't want to be looking at data from that camera anyway. (Maybe later down the line the object detection node can be modified to only look for world objects that are visible in this same cameras_used list...?)

Reply:
Ah, I see. Disabling image publishing for some camera doesn't prevent that camera from being used by all other on-board perception functionality. I agree we should ensure all frames are broadcasted.